lnclient/cashu/cashu.go

```
@@ -43,37 +45,31 @@ func NewCashuService(workDir string, mintUrl string) (result lnclient.LNClient,
 	logger.Logger.WithField("mintUrl", mintUrl).Info("Setting up cashu wallet")
 	config := wallet.Config{WalletPath: newpath, CurrentMintURL: mintUrl}
 
-	wallet, err := wallet.LoadWallet(config)
+	cashuWallet, err := wallet.LoadWallet(config)
 	if err != nil {
 		logger.Logger.WithError(err).Error("Failed to load cashu wallet")
 		return nil, err
 	}
 
 	cs := CashuService{
-		wallet: wallet,
+		wallet:  cashuWallet,
+		workDir: workDir,
 	}
 
-	// try to make an invoice to ensure the mint is running
-	// TODO: remove once LoadWallet is improved - see https://github.com/elnosh/gonuts/issues/49
-	_, err = cs.MakeInvoice(context.Background(), 10000, "", "", 0)
-	if err != nil {
-		logger.Logger.WithError(err).Error("Failed to load cashu wallet")
-		if isFirstSetup {
-			// delete wallet to ensure user gets a fresh start when they try a different mint
-			// otherwise the wallet freezes on next startup
-			fileErr := os.RemoveAll(newpath)
-			if fileErr != nil {
-				logger.Logger.WithError(fileErr).Error("Failed to remove broken wallet directory")
-			}
+	cashuWalletMnemonic := cashuWallet.Mnemonic()
+	mnemonic, _ := cfg.Get("Mnemonic", encryptionKey)
+	if mnemonic != cashuWalletMnemonic {
```

this will overwrite an existing mnemonic (there shouldn't be one). But should the check be `if mnemonic == ""` ?

---

Actually, if the mnemonic is different and not an empty string, we should probably return an error rather than overwriting it.

---

I was first checking if it was empty but while testing it seems a different (from cashu wallet's) mnemonic was being created and set so that's why I put that check. I'm assuming the different one was set in first startup.

---

So what does this mean for existing hub users? their old mnemonic will be overwritten, right?

---

looking at this: https://github.com/getAlby/hub/blob/master/service/keys/keys.go#L63-L75 it seems one is created anyways even if the mnemonic is set to false for any specific backend. So mnemonic would never be empty. Maybe this mnemonic should not be overwritten (?). I'm not sure what it's used for. Should I put the cashu wallet mnemonic in some other key?

---

Damn, I see. Ideally we use the one that Alby Hub generates as that's the one we ask the user to back up as part of the getting started steps, and then we pass that mnemonic to gonuts. Is this possible?

If gonuts always overwrites the mnemonic it is inconsistent with the rest of our LNClients and will cause other issues later on.

---

I noticed that "Backup" section in settings does not currently show up for cashu backend since mnemonic is set to false. 

> If gonuts always overwrites the mnemonic it is inconsistent with the rest of our LNClients and will cause other issues later on.

Ok, sure. 

Instead, maybe it can just be another command with that "Node Command" API you guys are planning? Send command to cashu wallet and that would return the mnemonic.

---

@elnosh I guess that's acceptable. Then we also need to undo the change to display that the cashu connector. I will add a comment. Thanks!

The commands PR is in progress and should be hopefully ready early next week: https://github.com/getAlby/hub/pull/1007

---

To move `cashuWallet.Mnemonic()` to a node command